golang/twinSender/main.go

```diff
@@ -0,0 +1,167 @@
+	}
+	log.Printf("Created Twin %s", twinMotionSensorIdentity.Did())
+
+	twinDescription, err := apiContext.TwinAPI.DescribeTwin(
+		apiContext.CtxWithMeta,
+		&ioticsApi.DescribeTwinRequest{
+			Headers: &apiContext.Headers,
+			Args: &ioticsApi.DescribeTwinRequest_Arguments{
+				TwinId: &ioticsApi.TwinID{
+					Id: twinMotionSensorIdentity.Did(),
+				},
+			},
+		},
+	)
+	if err != nil {
+		log.Fatalf("Could not Describe Twin: %v", err)
+	}
+	log.Printf("Twin description: %s", twinDescription.GetPayload())
+
+	/*** SEARCH FOR TWIN RADIATOR BY LABEL ***/
+	/* "We need to Search in the entire Network of Spaces (scope=GLOBAL)
+	   rather than locally (scope=LOCAL) in order to find Twins in a remote Host.
+	   The only search parameter we want to use for the sake of this exercise is the Twin's Label
+	   which must match exactly the Label of the Twin we want to find.
+	   -	Initially no Twins will be found (hostMetadataAllowList of Twin Radiator = noHost). */
+	headersSearch := ioticsApi.Headers{
+		ClientAppId:    "Iotics Search",
+		TransactionRef: []string{"Search"},
+		// Timeout of 3 seconds is enough given the poor amount of Twins in the Network
+		RequestTimeout: timestamppb.New(time.Now().Add(time.Second * 3)),
+	}
+
+	searchResponseClient, searchErr := apiContext.SearchAPI.SynchronousSearch(
+		apiContext.CtxWithMeta,
+		&ioticsApi.SearchRequest{
+			Headers: &headersSearch,
+			Scope:   ioticsApi.Scope_GLOBAL,
+			Payload: &ioticsApi.SearchRequest_Payload{
+				ResponseType: ioticsApi.ResponseType_FULL,
+				Filter: &ioticsApi.SearchRequest_Payload_Filter{
+					Properties: []*ioticsApi.Property{
+						{
+							Key: helpers.PROPERTY_KEY_LABEL,
+							Value: &ioticsApi.Property_LangLiteralValue{
+								LangLiteralValue: &ioticsApi.LangLiteral{
+									Value: "Twin Radiator", // BONUS: "Twin Thermostat"
+									Lang:  "en",
+								},
+							},
+						},
+					},
+				},
+			},
+		},
+	)
+	if searchErr != nil {
+		log.Fatalf("Could not search: %v", searchErr)
+	}
+
+	var twinsFound []*ioticsApi.SearchResponse_TwinDetails
+
+	for {
+		resp, respErr := searchResponseClient.Recv()
+		if respErr != nil {
+			break
+		}
+		twinsFound = append(twinsFound, resp.GetPayload().GetTwins()...)
+	}
+
+	log.Printf("Twins found: %s", twinsFound)
+
+	/*** SEND INPUT MESSAGES ***/
+	for {
+		randBool := rand.Intn(2) == 1                            // Generate random bool
+		inputMessage := fmt.Sprintf(`{"turn_on": %t}`, randBool) // BONUS: "presence_detected"
+		dataToShare := []byte(inputMessage)
+
+		_, err := apiContext.InterestAPI.SendInputMessage(
+			apiContext.CtxWithMeta,
+			&ioticsApi.SendInputMessageRequest{
+				Headers: &apiContext.Headers,
+				Args: &ioticsApi.SendInputMessageRequest_Arguments{
+					Interest: &ioticsApi.InputInterest{
+						SenderTwinId: &ioticsApi.TwinID{Id: twinMotionSensorIdentity.Did()},
+						DestInputId: &ioticsApi.InputID{
+							Id:     twinsFound[0].GetInputs()[0].InputId.Id,
```

Comment:
is it not possible to just set the whole InputID == twinsFound[0].GetInputs()[0].InputId?